File: js/euler63.js
function euler63(){
    var count = 0;
    // 10^n has n + 1 digits, so the base x can be at most 9 for x^n to have n digits
    for (let base = 1; base < 10; base++){
        let power = 1;
        let result = base;
        // increment count for every nth power found with n digits
        while (result.toString().length == power){
            count++;
            power++;
            // get next power by multiplying by base
            result *= base;
        }
    }
    return count;
}

console.log(euler63());
 

File: go/euler63.go
package main
import (
    "fmt"
    "math/big"
)

func main() {
    res := 0
    // nth power with n digits has a maximum base of 9, 10^n has (n - 1) digits
    for n := 1; n < 10; n++ {
        num := big.NewInt(1)
        base := big.NewInt(int64(n))
        power := 1
        for {
            num.Mul(num, base)
            if len(num.String()) != power {
                break
            }
            res++
            power++
        }
    }
    fmt.Println(res)
}
